test/cppsuite/tests/reverse_split.cpp

```diff
@@ -0,0 +1,168 @@
+        : operation_tracker(config, use_compression, tsm)
+    {
+    }
+
+    void
+    set_tracking_cursor(WT_SESSION *session, const tracking_operation &operation,
+      const uint64_t &collection_id, const std::string &key, const std::string &value,
+      wt_timestamp_t ts, scoped_cursor &op_track_cursor) override final
+    {
+        /* You can replace this call to define your own tracking table contents. */
+        operation_tracker::set_tracking_cursor(
+          session, operation, collection_id, key, value, ts, op_track_cursor);
+    }
+};
+
+/*
+ * This test inserts data at the end of the collection and truncates off from the start of the
+ * collection. In doing so pages at the start of the tree are gradually emptied while pages are
+ * added at the end of the tree. This means the test frequently executes the reverse split path.
+ */
+class reverse_split : public test {
+    public:
+    reverse_split(test_args &args) : test(args)
+    {
+        /* Add split timing stresses to the conn_open config. */
+        if (args.wt_open_config == "") {
+            std::string stress;
+            if (random_generator::instance().generate_bool())
+                stress = "timing_stress_for_test=[split_3]";
+            else
+                stress = "timing_stress_for_test=[split_4]";
+            logger::log_msg(LOG_WARN, "Adding config to WiredTiger open: " + stress);
+            args.wt_open_config = stress;
+        }
+        init_operation_tracker();
+    }
+
+    /*
+     * Insert operation that inserts a new K/V pair every op_rate.
+     */
+    void
+    insert_operation(thread_worker *tc) override final
+    {
+        logger::log_msg(
+          LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+        /* Must have unique collection for each thread. */
+        testutil_assert(tc->db.get_collection_count() == tc->thread_count);
+        collection &coll = tc->db.get_collection(tc->id);
+        scoped_cursor write_cursor = std::move(tc->session.open_scoped_cursor(coll.name));
+        while (tc->running()) {
+            tc->txn.begin();
+            int added_count = 0;
+            uint64_t start_key = coll.get_key_count();
+            while (tc->txn.active()) {
+                auto key = tc->pad_string(std::to_string(start_key + added_count), tc->key_size);
+                auto value =
+                  random_generator::instance().generate_pseudo_random_string(tc->value_size);
+                /* A return value of true implies the insert was successful. */
+                if (!tc->insert(write_cursor, coll.id, key, value)) {
+                    tc->txn.rollback();
+                    break;
+                }
+                added_count++;
+                if (tc->txn.can_commit()) {
+                    if (tc->txn.commit()) {
+                        coll.increase_key_count(added_count);
+                        start_key = coll.get_key_count();
```

Comment:
Do you need this? As soon as you commit, `tc->txn.active()` is false and the `while` loop will exit.

Reply:
I think you reviewed an older version, this code is removed in later commits.